src/tests/redecomp_sparsematrix.cpp

```diff
@@ -0,0 +1,223 @@
+    auto filter_radius = std::get<2>(GetParam());  // filter radius of the explicit density filter
+    
+    double side_length = 1.0; // side length of square domain
+    int n_elem_per_dim = 4;   // number of elements per dimension
+    mfem::Mesh serial_mesh;
+    if (dim == 2) {
+      serial_mesh = mfem::Mesh::MakeCartesian2D(n_elem_per_dim, n_elem_per_dim, 
+        mfem::Element::Type::QUADRILATERAL, false, side_length, side_length);
+    }
+    else {
+      serial_mesh = mfem::Mesh::MakeCartesian3D(n_elem_per_dim, n_elem_per_dim, n_elem_per_dim, 
+        mfem::Element::Type::HEXAHEDRON, side_length, side_length, side_length);
+    }
+    // refine mesh "ref_levels" times
+    for (int i{0}; i < ref_levels; ++i){
+      serial_mesh.UniformRefinement();
+    }
+    // update the number of elements per dimension to reflect the refined mesh
+    n_elem_per_dim *= std::pow(2, ref_levels);
+    // compute the element side length
+    double dx = side_length / n_elem_per_dim;
+
+    // build a parallel mesh with associated 0-order, L2 finite element space
+    mfem::ParMesh par_mesh { MPI_COMM_WORLD, serial_mesh };
+    mfem::L2_FECollection l2_elems { 0, dim };
+    mfem::ParFiniteElementSpace par_fes { &par_mesh, &l2_elems };
+
+    // create RedecompMesh and fe space and sparse matrix transfer object
+    redecomp::RedecompMesh redecomp_mesh { par_mesh, redecomp::RedecompMesh::RCB, filter_radius };
+    mfem::FiniteElementSpace redecomp_fes { &redecomp_mesh, &l2_elems };
+    redecomp::SparseMatrixTransfer matrix_xfer {
+      par_fes,
+      par_fes,
+      redecomp_fes, 
+      redecomp_fes
+    };
+
+    // compute sparse matrix on redecomp_fes (rows and col)
+    mfem::SparseMatrix W_redecomp { redecomp_fes.GetVSize(), redecomp_fes.GetVSize() };
+
+    // define kernel of density filter (using isotropic "cone" filter here)
+    auto filter_kernel = [&filter_radius](const mfem::Vector &xi, const mfem::Vector &xj) {
+      return std::max(0.0, filter_radius - xi.DistanceTo(xj));
+    };
+
+    // intialize containers needed for filter evaluation
+    int n_local_elem = redecomp_mesh.GetNE();
+    mfem::Vector xi(dim), xj(dim);
+    std::vector<int> n_row_entries(n_local_elem);
+
+    // loop over each element to form a row of the matrix
+    for (int i=0; i<n_local_elem; i++) {
+      redecomp_mesh.GetElementCenter(i, xi);
+
+      // loop over each element to fill in each column (j) of row (i)
+      n_row_entries[i] = 0;
+      for (int j=0; j<n_local_elem; j++) {
+        redecomp_mesh.GetElementCenter(j, xj);
+        // evaluate kernel which takes in both element centroids
+        double Wij = filter_kernel(xi, xj);
+        // if kernel returns a non-zero value, create entry in the sparse redecomp matrix
+        if (Wij > 0.0) {
+          W_redecomp.Add(i, j, Wij);
+          n_row_entries[i]++;
+        }
+      }
+    }
+
+    W_redecomp.Finalize();
```

Comment:
Consider noting that this is required in order to interface with the redecomp capability.